File: backend-api/AwardSystemAPI.Domain/Entities/AwardProcess.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AwardSystemAPI.Domain.Entities
{
    [Table("awardProcess")]
    public class AwardProcess
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(255)]
        public required string AwardsName { get; set; }

        [Required]
        public DateTime StartDate { get; set; }

        public DateTime EndDate { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    }
}

File: web-app/award-web/src/types/AwardProcess.ts
export interface AwardProcess {
    id: number;
    awardsName: string;
    startDate: string;
    endDate: string;
    status: string;
    createdAt: string;
}

export interface CreateAwardProcessPayload {
    awardsName: string;
    startDate: string;
    endDate: string;
    status: string;
}

export interface JudgingRound {
    id: number;
    awardProcessId: number;
    roundName: string;
    startDate: string;
    deadline: string;
    candidateCount: number;
    createdAt: string;
    updatedAt: string;
}

export interface CreateJudgingRoundPayload {
    awardProcessId: number;
    roundName: string;
    startDate: string;
    deadline: string;
    candidateCount: number;
}
